### PR TITLE
samples: mgmt: smp_svr: Add Serial over USB CDC_ACM overlay.

### PR DESCRIPTION
**Pull request by eobalski**
This patch adds overlay for USB serial backend over CDC_ACM
device.

Signed-off-by: Emil Obalski <emil.obalski@nordicsemi.no>

### CODE REVIEW

**Comment by de-nordic** [comment]
@emob-nordic  Does this require https://github.com/zephyrproject-rtos/zephyr/pull/28399 and https://github.com/zephyrproject-rtos/zephyr/pull/28408 to work?

**Comment by emob-nordic** [comment]
> @emob-nordic Does this require #28399 and #28408 to work?

#28399 yes, #28408 no -> just for increased performance.
Anyway this is just an overlay to default configuration so I've removed those dependencies.

**Review by pabigot — CHANGES_REQUESTED** [review]
This works for me with the other PRs merged in.  Could you add the following to `sample.yaml` so we get build-tests of this in CI?
```
  sample.mcumg.smp_svr.cdc:
    extra_args: OVERLAY_CONFIG="overlay-cdc.conf"
    platform_allow: nrf52dk_nrf52832 nrf52840dk_nrf52840
```

**Comment by emob-nordic** [comment]
> This works for me with the other PRs merged in. Could you add the following to `sample.yaml` so we get build-tests of this in CI?
> 
> ```
>   sample.mcumg.smp_svr.cdc:
>     extra_args: OVERLAY_CONFIG="overlay-cdc.conf"
>     platform_allow: nrf52dk_nrf52832 nrf52840dk_nrf52840
> ```

Will add 5340 as well.

**Comment by nvlsianpu** [comment]
looks good

**Review by pabigot — APPROVED** [review]

**Review by nvlsianpu — APPROVED** [review]

File: samples/subsys/mgmt/mcumgr/smp_svr/sample.yaml [hunk 1]
```diff
@@ -14,3 +14,6 @@ tests:
   sample.mcumg.smp_svr.udp:
     extra_args: OVERLAY_CONFIG="overlay-udp.conf"
     platform_allow: frdm_k64f
+  sample.mcumg.smp_svr.cdc:
+    extra_args: OVERLAY_CONFIG="overlay-cdc.conf"
+    platform_allow: nrf52dk_nrf52832 nrf52840dk_nrf52840 nrf5340dk_nrf5340_cpuapp
```
Comment (jfischer-phytec-iot) [inline samples/subsys/mgmt/mcumgr/smp_svr/sample.yaml:19]: nrf52dk_nrf52832 ? It does not have USB device controller.

**Review by jfischer-no — CHANGES_REQUESTED** [review]

**Review by de-nordic — APPROVED** [review]
Looks OK.

Built, flashed, run and tested with https://github.com/zephyrproject-rtos/zephyr/pull/28399 applied on top on nrf52840dk, slowly :) but works.

**Comment by emob-nordic** [comment]
> slowly :) but works.

slowly but carefully... :laughing:

**Review by jfischer-no — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**